src/utils/parsers.rs

```diff
@@ -60,6 +61,47 @@ pub fn parse_graph_ref(graph_id: &str) -> Result<GraphRef> {
     }
 }
 
+#[derive(Debug, Serialize, Default, Clone)]
+pub struct ValidationPeriod {
+    pub from: Option<String>,
+    pub to: Option<String>,
+}
+
+/// Validation period is a positive number of seconds to validate in the past.
+// We just need to validate and negate it
+pub fn parse_validation_period(period: &str) -> Result<ValidationPeriod> {
+    let window = period.parse::<i64>()?;
+    if window > 0 {
+        Ok(ValidationPeriod {
+            from: Some(format!("{}", -window)),
+            to: Some("-1".to_string()),
+        })
+    } else {
+        Err(
+            anyhow!("Invalid validation period. Must be a positive number of seconds.".to_string())
+                .into(),
+        )
+    }
+}
+
+pub fn parse_query_count_threshold(threshold: &str) -> Result<i64> {
+    let threshold = threshold.parse::<i64>()?;
+    if threshold < 1 {
+        Err(anyhow!("Invalid value for query count threshold. Must be a positive integer.").into())
+    } else {
+        Ok(threshold)
+    }
+}
+
+pub fn parse_query_percentage_threshold(threshold: &str) -> Result<f64> {
+    let threshold = threshold.parse::<i64>()?;
+    if threshold <= 0 || threshold >= 100 {
```

Comment:
can it not be exactly 100 or exactly 0?

Reply:
Ah, it looks like it can be actually! Good catch